akka-persistence-typed/src/main/scala/akka/persistence/typed/javadsl/ActiveActiveEventSourcedBehavior.scala

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (C) 2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.persistence.typed.javadsl
+
+import java.util.Optional
+
+import akka.actor.typed.BackoffSupervisorStrategy
+import akka.actor.typed.Behavior
+import akka.actor.typed.TypedActorContext
+import akka.annotation.InternalApi
+import akka.persistence.typed.internal
+import akka.persistence.typed.internal.EffectImpl
+import akka.persistence.typed.scaladsl.ActiveActiveContextImpl
+
+abstract class ActiveActiveEventSourcedBehavior[Command, Event, State](
+    activeActiveContext: ActiveActiveContext,
+    onPersistFailure: Optional[BackoffSupervisorStrategy])
+    extends EventSourcedBehavior[Command, Event, State](activeActiveContext.persistenceId, onPersistFailure) {
+
+  def this(activeActiveContext: ActiveActiveContext) = this(activeActiveContext, Optional.empty())
+
+  protected def getActiveActiveContext(): ActiveActiveContext = activeActiveContext
+
+  /**
+   * INTERNAL API: DeferredBehavior init, not for user extension
+   */
+  @InternalApi override def apply(context: TypedActorContext[Command]): Behavior[Command] = {
+    // Note: duplicated in EventSourcedBehavior to not break source compatibility
```

Comment:
i guess it would not be nice to pull this out to a private[Akka] internal method as well as java users will see it

Reply:
Yeah, I tried that as well as putting it in a static internal elsewhere but it was kind of nasty for such a small gain, I don't expect this method will change much over time anyway.